src/components/vm/consoles/vnc.jsx

```diff
@@ -404,17 +428,29 @@ export class VncActive extends React.Component {
             this.observer.disconnect();
     }
 
-    onDisconnected(detail) { // server disconnected
-        console.info('Connection lost: ', detail);
-        this.props.state.setConnected(false);
+    async onDisconnected(event) { // server disconnected
+        console.info('Connection lost: ', event.detail);
+        let reason = this.props.state.failure_reason;
+        if (event.detail.clean)
+            reason = null;
+        else if (!reason) {
+            // This might be TLS.
+            const qemu_conf = await readQemuConf();
+            if (qemu_conf.vnc_tls)
+                reason = _("VNC with TLS is not supported by the in-page viewer");
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2038-5fe3bc4a-20250808-144213-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)